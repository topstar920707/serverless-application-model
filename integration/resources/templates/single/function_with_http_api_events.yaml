Resources:
  MyHttpApi:
    Type: AWS::Serverless::HttpApi
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs12.x
      CodeUri: ${codeuri}
      Events:
        FooEvent:
          Type: HttpApi
          Properties:
            ApiId:
              Ref: MyHttpApi
        BarEvent:
          Type: HttpApi
          Properties:
            ApiId:
              Ref: MyHttpApi
            Path: /bar
            Method: POST
