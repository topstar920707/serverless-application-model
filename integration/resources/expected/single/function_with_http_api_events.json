[
  { "LogicalResourceId":"MyLambdaFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyLambdaFunctionFooEventPermission", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"MyHttpApi", "ResourceType":"AWS::ApiGatewayV2::Api" },
  { "LogicalResourceId":"MyHttpApiApiGatewayDefaultStage", "ResourceType":"AWS::ApiGatewayV2::Stage" }
]
